common/base/include/claragenomics/utils/allocator.hpp

```diff
@@ -230,19 +230,21 @@ class CachingDeviceAllocator
 
     /// \brief asynchronously allocates a device array with enough space for n elements of value_type
     /// \param n number of elements to allocate the array for
-    /// \param stream CUDA stream to be associated with this method
+    /// \param streams CUDA streams to be associated with this memory allocation, when memory allocation is deallocated cudaStreamSynchronize() is called on all of them
     /// \return pointer to allocated memory
     /// \throw device_memory_allocation_exception if allocation was not successful
-    pointer allocate(std::size_t n, cudaStream_t stream = 0)
+    pointer allocate(std::size_t n, const std::vector<cudaStream_t>& streams = {{0}})
```

Comment:
I think we don't need to provide a default argument for the stream parameter here. This should be done further up. Furthermore, the CudaMallocAllocator and this class should have the same interface and the CudaMallocAllocator does not provide a default argument either.

Reply:
We do need it in order to be able to use it in thurst. Thrust calls custom allocator's `allocate(std::size n)`. As it does not provide any stream it should sync on the default stream.

This means actual bug actually me not including default value in `CudaMallocAllocator` as well